File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/relationships/cs/workwithrelationships.retrieverelationship.cs
// <snippetworkwithrelationships.retrieverelationship>



 //You can use either the Name or the MetadataId of the relationship.

 //Retrieve the One-to-many relationship using the MetadataId.
 RetrieveRelationshipRequest retrieveOneToManyRequest =
     new RetrieveRelationshipRequest { MetadataId = _oneToManyRelationshipId };
 RetrieveRelationshipResponse retrieveOneToManyResponse =
     (RetrieveRelationshipResponse)_serviceProxy.Execute(retrieveOneToManyRequest);

 Console.WriteLine("Retrieved {0} One-to-many relationship by id", retrieveOneToManyResponse.RelationshipMetadata.SchemaName);

 //Retrieve the Many-to-many relationship using the Name.
 RetrieveRelationshipRequest retrieveManyToManyRequest =
     new RetrieveRelationshipRequest { Name = _manyToManyRelationshipName};
 RetrieveRelationshipResponse retrieveManyToManyResponse =
     (RetrieveRelationshipResponse)_serviceProxy.Execute(retrieveManyToManyRequest);

 Console.WriteLine("Retrieved {0} Many-to-Many relationship by Name", retrieveManyToManyResponse.RelationshipMetadata.MetadataId);

// </snippetworkwithrelationships.retrieverelationship>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/relationshipsvb/vb/workwithrelationships.retrieverelationship.vb
' <snippetworkwithrelationships.retrieverelationship>



 'You can use either the Name or the MetadataId of the relationship.

 'Retrieve the One-to-many relationship using the MetadataId.
 Dim retrieveOneToManyRequest As RetrieveRelationshipRequest =
  New RetrieveRelationshipRequest With {.MetadataId = _oneToManyRelationshipId}
 Dim retrieveOneToManyResponse As RetrieveRelationshipResponse =
  CType(_serviceProxy.Execute(retrieveOneToManyRequest), RetrieveRelationshipResponse)

 Console.WriteLine("Retrieved {0} One-to-many relationship by id",
                   retrieveOneToManyResponse.RelationshipMetadata.SchemaName)

 'Retrieve the Many-to-many relationship using the Name.
 Dim retrieveManyToManyRequest As RetrieveRelationshipRequest =
  New RetrieveRelationshipRequest With {
   .Name = _manyToManyRelationshipName
  }
 Dim retrieveManyToManyResponse As RetrieveRelationshipResponse =
  CType(_serviceProxy.Execute(retrieveManyToManyRequest), RetrieveRelationshipResponse)

 Console.WriteLine("Retrieved {0} Many-to-Many relationship by Name",
                   retrieveManyToManyResponse.RelationshipMetadata.MetadataId)

' </snippetworkwithrelationships.retrieverelationship>